metadata-ingestion/src/datahub/ingestion/source/openapi.py

```diff
@@ -61,11 +62,22 @@ class OpenApiConfig(ConfigModel):
     )
     forced_examples: dict = Field(default={}, description="")
     token: Optional[str] = Field(default=None, description="")
+    bearer_token: Optional[str] = Field(default=None, description="Use Bearer token")
     get_token: dict = Field(default={}, description="")
 
+    @validator("bearer_token")
+    def ensure_only_one_token(
+        cls, bearer_token: Optional[str], values
+    ) -> Optional[str]:
+        if bearer_token is not None and values.get("token") is not None:
+            raise ConfigurationError(
+                "Unable to use 'token' and 'bearer_token' together."
+            )
+        return bearer_token
+
     def get_swagger(self) -> Dict:
-        if self.get_token or self.token is not None:
-            if self.token is not None:
+        if self.get_token or self.token or self.bearer_token is not None:
+            if self.token or self.bearer_token:
```

Comment:
I was imagining something like this

It might be good to also move the token resolution logic to a separate method

```suggestion
            if self.token:
                pass
            elif self.bearer_token:
                self.token = f"Bearer {self.bearer_token}"
```